infra/src/test/kotlin/fi/fta/geoviite/infra/ui/pagemodel/map/ToolPanelDialog.kt

```diff
@@ -0,0 +1,166 @@
+package fi.fta.geoviite.infra.ui.pagemodel.map
+
+import fi.fta.geoviite.infra.ui.pagemodel.common.E2EDialog
+import fi.fta.geoviite.infra.ui.pagemodel.common.defaultDialogBy
+import org.openqa.selenium.By
+import waitUntilDoesNotExist
+
+class E2ELocationTrackEditDialog(by: By = defaultDialogBy) : E2EDialog(by) {
+    enum class State(val uiText: String) {
+        IN_USE("Käytössä"),
+        NOT_IN_USE("Käytöstä poistettu"),
+        DELETED("Poistettu")
+    }
+
+    enum class Type(val uiText: String) {
+        MAIN("Pääraide"),
+        SIDE("Sivuraide"),
+        TRAP("Turvaraide"),
+        CHORD("Kujaraide"),
+    }
+
+    enum class TopologicalConnectivity(val uiText: String) {
+        NONE("Ei kytketty"),
+        START("Raiteen alku"),
+        END("Raiteen loppu"),
+        START_AND_END("Raiteen alku ja loppu")
+    }
+
+    fun setName(name: String): E2ELocationTrackEditDialog = apply {
+        content.inputFieldValue("Sijaintiraidetunnus", name)
+    }
+
+    fun selectTrackNumber(trackNumber: String): E2ELocationTrackEditDialog = apply {
+        content.selectDropdownValue("Ratanumero", trackNumber)
+    }
+
+    fun selectState(state: State): E2ELocationTrackEditDialog = apply {
+        content.selectDropdownValue("Tila", state.uiText)
+    }
+
+    fun selectType(type: Type): E2ELocationTrackEditDialog = apply {
+        content.selectDropdownValue("Raidetyyppi", type.uiText)
+    }
+
+    fun setDescription(description: String): E2ELocationTrackEditDialog = apply {
+        content.inputFieldValue("Kuvaus", description)
+    }
+
+    fun selectTopologicalConnectivity(topologicalConnectivity: TopologicalConnectivity): E2ELocationTrackEditDialog =
+        apply {
+            content.selectDropdownValue(label = "Topologinen kytkeytyminen", topologicalConnectivity.uiText)
+        }
+
+    fun save() {
+        val element = webElement
+        val isDeleted = content.getValueForField("Tila") == State.DELETED.uiText
+        clickButtonByText("Tallenna")
+
+        if (isDeleted) {
+            childElement(
+                By.xpath("div[contains(@class, 'dialog')]//button[contains(@class, 'button--primary')]")
```

Comment:
Yritin yhtenäistää sisäkkäisten dialogien käsittelyä, mutta joissain tilanteissa ne tulivat rinnalle eikä sisäkkäin dom-puussa.